syftlib/src/main/java/org/openmined/syft/execution/JobDownloader.kt

```diff
@@ -0,0 +1,181 @@
+package org.openmined.syft.execution
+
+import android.util.Log
+import io.reactivex.Single
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.processors.PublishProcessor
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.networking.datamodels.ClientConfig
+import org.openmined.syft.proto.SyftModel
+import org.openmined.syft.utilities.FileWriter
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicReference
+
+private const val TAG = "JobDownloader"
+
+@ExperimentalUnsignedTypes
+internal class JobDownloader {
+    fun downloadData(
+        workerId: String,
+        config: SyftConfiguration,
+        requestKey: String?,
+        networkDisposable: CompositeDisposable,
+        jobStatusProcessor: PublishProcessor<JobStatusMessage>,
+        clientConfig: ClientConfig?,
+        plans: ConcurrentHashMap<String, Plan>,
+        model: SyftModel,
+        protocols: ConcurrentHashMap<String, Protocol>
+    ) {
+        val trainingParamsStatus = AtomicReference(SyftJob.DownloadStatus.NOT_STARTED)
```

Comment:
This seems wierd. It should be state of syftjob so that we don't download again if previous download is in progress

Reply:
It doesn't belong here, does it? However, as you can see, it's not referenced outside it.
Once the tests are done, I plan to do more work on refactoring `JobDownloader`.

Reply:
It should be a property of job

Reply:
Exposed it for `Job` to check on its status.